pkg/cmd/util/clientcmd/factory.go

```diff
@@ -166,6 +169,31 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 	w.Printer = func(mapping *meta.RESTMapping, noHeaders, withNamespace, wide bool, showAll bool, columnLabels []string) (kubectl.ResourcePrinter, error) {
 		return describe.NewHumanReadablePrinter(noHeaders, withNamespace, wide, showAll, columnLabels), nil
 	}
+	kAttachablePodForObjectFunc := w.Factory.AttachablePodForObject
+	w.AttachablePodForObject = func(object runtime.Object) (*api.Pod, error) {
+		_, kc, err := w.Clients()
+		if err != nil {
+			return nil, err
+		}
+		switch t := object.(type) {
+		case *deployapi.DeploymentConfig:
+			var pods *api.PodList
```

Comment:
I'm not confident it should be the deployer pod. If you do `oc run nginx --image=nginx --attach` aren't you expecting to attach to the actual nginx pod? In fact you may not even know a deployment is involved in this `oc run` operation. @liggitt opinion?